### DISCUSSION TITLE
Updating field values

### DISCUSSION THREAD

**Discussion by bkilshaw** [naymspace/backpex#482] [Q&A]
I've been playing around with creating a custom field that lets a user search for an address, it uses the google auto complete API to suggest matches, and when the user selects a result it auto-populates the related address fields.

Are there any existing methods available that would let me set the value of a various fields outside the normal flow?

```elixir
 def handle_event("select", %{"value" => place_id}, socket) do
   address_details = get_address_details(place_id)

    # set address related fields

    {:noreply, socket}
  end
  ```

**Comment by Flo0807** [comment]
Hey, I appreciate your patience with this!

It is possible to set values in the changeset function. In this function you can also access the `assigns` and `target` inputs.

Here is some example code:

```elixir
def changeset(user, attrs, metadata) do
  # the metadata contains the target input and assigns
  target = Keyword.get(metadata, :target)
  assigns = Keyword.get(metadata, :assigns)

  user
  |> cast(attrs, [...])
  |> maybe_put_address_details(target)
  ...
end

# this function is being called if the input target is "place"
def maybe_put_address_details(changeset, "place" = _target) do
  place_id = get_field(changeset, :place)
  address_details = get_address_details(place_id)

  put_change(changeset, :address_details, address_details)
end

def maybe_put_address_details(changeset, _target) do
  changeset
end
```
